crypto/keys/ed25519/ed25519_test.go

```diff
@@ -228,3 +231,80 @@ func TestMarshalAmino_BackwardsCompatibility(t *testing.T) {
 		})
 	}
 }
+
+// TODO - maybe we should move the tests below to `codec_test` package, WDYT?
```

Comment:
Sounds like a good idea 👍

Reply:
So, I'm not convinced any more (doesn't have a preference). This tests only deals with ed25519. On the other hand it's a good example for showing how to deal with interface encoding. 

Other idea is to link this test in a documentation. 

Anyone has a preference?

Reply:
ed25519 is just used as an example here so I think I'd prefer to put it in codec_test, although not a strong preference.

Reply:
OK, then let's do that. Moved to `/codec`